### ISSUE TITLE
Upgrade Ubuntu Base Image to 22.10 to address Trivy Security Vulnerabilities

### ISSUE THREAD

**Issue by Darnesey** [envoyproxy/envoy#22587]
Trivy has reported several vulnerabilities for envoy images, but they have primarily only existed within the base images the envoy binaries are built on. Currently from my observations of your dockerhub envoyproxy profile, you have 3 main versions you support*:
- envoy (on Ubuntu)
- envoy-alpine (*supposedly will be dropping this)
- envoy-distroless (debian)

I see that you have upgraded to Ubuntu:22.04 recently ([PR Link](https://github.com/envoyproxy/envoy/pull/21388)), and I wish I had seen it earlier as I would have just commented on that PR instead.  

Unfortunately Debian does not appear to have any standard release builds (eg. 11.4) that don't have reported vulnerabilities. Some other distroless images from other libraries have used a custom Debian build to get around that but I'm not sure how well those are maintained.

## Resolution Strategy
Please update `envoyproxy/envoy`'s base image to `ubuntu:22.10`.

### Trivy Vulnerability Summary Count Based on Image
`envoyproxy/envoy:v1.22.4 (ubuntu 18.04) -- Total: 37 (UNKNOWN: 0, LOW: 32, MEDIUM: 5, HIGH: 0, CRITICAL: 0)`
`envoyproxy/envoy-distroless:v1.22.4 (debian 11.4) -- Total: 14 (UNKNOWN: 0, LOW: 11, MEDIUM: 0, HIGH: 3, CRITICAL: 0)`
`ubuntu:22.04 -- Total: 17 (UNKNOWN: 0, LOW: 15, MEDIUM: 2, HIGH: 0, CRITICAL: 0)`
`ubuntu:22.10 -- Total: 0`
`debian:11.4 -- Total: 14 (UNKNOWN: 0, LOW: 11, MEDIUM: 0, HIGH: 3, CRITICAL: 0)`

**Comment by phlax** [comment]
one question i have about how you are running trivy 

as i understand it the recommended way is to ignore bugs with no fixes - on the basis that there is nothing that you can do about it - im guessing that is not how you are running it

**Comment by Darnesey** [comment]
That is a good point. We are technically not using that flag and I can talk with my security managers around whether we want to use that flag at all (there's still value in being aware of unfixed vulnerabilities in our reports). There's only one vulnerability on 22.04 that has a fix now. Generally we like to be aware of unfixed vulnerabilities in case there's a newer base image version which does not contain those vulnerabilities. 22.10 seemed to suggest that.

That said, I just realized Trivy doesn't support scanning the 22.10 image yet (didn't realize 22.10 doesn't technically release until October 2022) so we can't even guarantee this image doesn't contain the same vulnerabilities. Apologies. Will close this issue and look out for updates when Trivy supports 22.10

**Comment by phlax** [comment]
for reference my PR to add trivy to Envoy CI is here #20779 - i have been running it by hand

not sure if i have the trivy settings optimally configured etc

**Comment by Darnesey** [comment]
From the looks of it, your trivy scans are only flagging vulnerabilities with the HIGH/CRITICAL severity levels. This is certainly fine if you're using it to block pipelines. I don't know what your process is for assigning risk levels for security bugs or bugs in general, but it might be useful to have a separate (unblocking) scan for the remaining severity levels. That way, this information is known when working on upgrading base images.

**Comment by phlax** [comment]
> This is certainly fine if you're using it to block pipelines.

its more scheduled/notification pipelines - but yep - i think we are mostly interested in the high/critical issues

>  I don't know what your process is for assigning risk levels for security bugs or bugs in general

well given that these issues are (afaiaa) ones that dont affect us - the issue is more just having the vulnerable software in the container, unused, these are generally regarded as low priority - in fact it wasnt something we tracked at all until recently

> That way, this information is known when working on upgrading base images.

perhaps, but i think when deciding on upgrading eg the ubuntu container, compatibility is a higher concern

**Comment by phlax** [comment]
> Compatibility should not be a big concern

glibc is the key compatibility issue

**Comment by phlax** [comment]
also, to be clear, its not too much of a concern

if you use the container with the envoy binary as is - none (afaiaa) of the issues that trivy has flagged are relevant

if you use the container with some other binary that does use those libs it would be a concern - but unless you are rebuilding Envoy in some way, you would already be using it in a way that was not intended and/or defies best practice

**Comment by Darnesey** [comment]
Yeah I figured it shouldn't matter too much so long as no binary is leveraging those packages and there's not any way for a user to access the CLI for that container.
